tools/src/main/java/org/apache/kafka/tools/ConnectPluginPath.java

```diff
@@ -0,0 +1,559 @@
+        boolean hasManifest = manifests.containsKey(className) && manifests.get(className).stream().anyMatch(e -> e.type == type);
+        return new Row(pluginLocation, className, type, version, rowAliases, loadable, hasManifest);
+    }
+
+    private static void beginCommand(Config config) {
+        if (config.command == Command.LIST) {
+            // The list command prints a TSV-formatted table with details of the found plugins
+            // This is officially human-readable output with no guarantees for backwards-compatibility
+            // It should be reasonably easy to parse for ad-hoc scripting use-cases.
+            listTablePrint(config, LIST_TABLE_COLUMNS);
+        }
+    }
+
+    private static void handlePlugin(Config config, Row row) {
+        if (config.command == Command.LIST) {
+            String firstAlias = row.aliases.size() > 0 ? row.aliases.get(0) : NO_ALIAS;
+            String secondAlias = row.aliases.size() > 1 ? row.aliases.get(1) : NO_ALIAS;
+            listTablePrint(config,
+                    row.className,
+                    firstAlias,
+                    secondAlias,
+                    row.version,
+                    row.type,
+                    row.loadable,
+                    row.hasManifest,
+                    // last because it is least important and most repetitive
+                    row.locationString()
+            );
+        }
+    }
+
+    private static void endCommand(
+            Config config,
+            Map<Path, Set<Row>> rowsByLocation
+    ) {
+        if (config.command == Command.LIST) {
+            // end the table with an empty line to enable users to separate the table from the summary.
+            config.out.println();
+            Set<Row> allRows = rowsByLocation.values().stream().flatMap(Set::stream).collect(Collectors.toSet());
+            Map<String, Set<String>> aliasCollisions = aliasCollisions(allRows);
+            for (Map.Entry<String, Set<String>> entry : aliasCollisions.entrySet()) {
+                String alias = entry.getKey();
+                Set<String> classNames = entry.getValue();
+                if (classNames.size() != 1) {
+                    config.out.printf("%s is an ambiguous alias and will not be usable. One of the fully qualified classes must be used instead:%n", alias);
+                    for (String className : classNames) {
+                        config.out.printf("\t%s%n", className);
+                    }
+                }
+            }
+            Set<Row> classpathOnlyPlugins = findClasspathOnlyPlugins(allRows);
+            Set<String> classNames = classpathOnlyPlugins.stream().filter(Row::incompatible).map(row -> row.className).collect(Collectors.toSet());
+            if (classNames.size() != 0) {
+                config.out.printf("%d plugins on the classpath are not compatible, and will not be auto-migrated. " +
```

Comment:
"will not" seems misleading; with the `list` command, nothing gets migrated at all.

Maybe "will not be eligible for auto-migration"? Although even that feels like too much, since (IIUC) if the user is adding plugins to their worker's classpath, it's viable for them to run this script on those plugins without that modified classpath.

Reply:
You're right that this comment is only relevant to sync-manifests. I'll remove this from this PR and we can consider adding it later.